CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading.Tasks;
+using CribblyBackend.Controllers;
+using CribblyBackend.DataAccess.Models;
+using CribblyBackend.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Serilog;
+using Xunit;
+
+namespace CribblyBackend.UnitTests
+{
+    public class DivisionControllerTests
+    {
+        private readonly Mock<HttpRequest> mockHttpRequest;
+        private readonly Mock<HttpContext> mockHttpContext;
+        private readonly DivisionController divisionController;
+        private readonly Mock<IDivisionService> mockDivisionService;
+        private readonly Mock<ILogger> mockLoggerService;
+
+
+        public DivisionControllerTests()
+        {
+            mockDivisionService = new Mock<IDivisionService>();
+            mockLoggerService = new Mock<ILogger>();
+            mockHttpRequest = new Mock<HttpRequest>();
+            mockHttpContext = new Mock<HttpContext>();
+            mockHttpContext.Setup(x => x.Request).Returns(mockHttpRequest.Object);
+            divisionController = new DivisionController(mockDivisionService.Object, mockLoggerService.Object);
+            divisionController.ControllerContext = new ControllerContext()
+            {
+                HttpContext = mockHttpContext.Object
+            };
+        }
+
+        [Fact]
+        public async Task Create_ShouldReturnOkAndCreatedId_IfNoError()
+        {
+            Division division = new Division();
+            mockDivisionService.Setup(x => x.Create(It.IsAny<Division>())).ReturnsAsync(division);
```

Comment:
You might want to set this up to return a division with an ID so you can make sure the ID gets populated

Reply:
How does that look?

Reply:
```
var division = new Division
{
    Id = "gettin jiggy wit it",
};
mockDivisionService.Setup(...).ReturnsAsync(division);
var result = await divisionController.Create(new Division());
var typedResult = Assert.IsType<OkObjectResult>(result);
var div = Assert.IsType<Division>(typedResult.Value);
Assert.Equal(div.Id, "gettin jiggy wit it");

Reply:
Now that I'm looking at that, though, this probably isn't the place to test for that -- you'd want to do something like this in the repository or service

Reply:
Done!